src/client/Microsoft.Identity.Client/PlatformsCommon/Shared/KeyedSemaphorePool.cs

```diff
@@ -0,0 +1,39 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
```

Comment:
If you remove a semaphore from the dictionary, should you also dispose it to avoid leaks?

```
if (_gates.TryRemove(key, out var removed))
{
    removed.Dispose();
}
```

Reply:
In the current design we no longer remove semaphores from the dictionary; each key uses a single SemaphoreSlim for the lifetime of the pool. That avoids races where a semaphore could be disposed while another caller still holds a reference. We also don’t touch AvailableWaitHandle, so relying on GC for clean‑up is sufficient for our usage.